mods/player_api/api.lua

```diff
@@ -60,37 +58,46 @@ function player_api.set_model(player, model_name)
 			eye_height = 1.625,
 		})
 	end
-	player_model[name] = model_name
+	player_data.model = model_name
 end
 
 function player_api.set_textures(player, textures)
-	local name = player:get_player_name()
-	local model = models[player_model[name]]
-	local model_textures = model and model.textures or nil
-	player_textures[name] = textures or model_textures
-	player:set_properties({textures = textures or model_textures})
+	local player_data = players[player:get_player_name()]
+	local model = models[player_data.model]
+	local new_textures = model and model.textures or textures
+	player_data.textures = new_textures
+	player:set_properties({textures = new_textures})
 end
 
 function player_api.set_animation(player, anim_name, speed)
-	local name = player:get_player_name()
-	if player_anim[name] == anim_name then
+	local player_data = players[player:get_player_name()]
+	local model = models[player_data.model]
+	if not (model and model.animations[anim_name]) then
 		return
 	end
-	local model = player_model[name] and models[player_model[name]]
-	if not (model and model.animations[anim_name]) then
+	speed = speed or model.animation_speed
+	if player_data.animation == anim_name and player_data.animation_speed == speed then
 		return
 	end
 	local anim = model.animations[anim_name]
-	player_anim[name] = anim_name
-	player:set_animation(anim, speed or model.animation_speed, animation_blend)
+	player_data.animation = anim_name
```

Comment:
`player_data` is local to the function, surely this needs to write data to `players[player_name]`?

Reply:
See above.